src/cpu/Core.cpp

```diff
@@ -240,8 +239,8 @@ void Core::instr_reti() {
 }
 
 void Core::instr_rst(Byte addr) {
-  instr_push(_pc.word + 2);
-  _pc.word = addr;
+  instr_push(_pc.word + 1);
+  _pc.word = addr - 1;
```

Comment:
The problem is that pc is incremented by one after the rst execution but should not.
_in_jump_state is made to avoid this problem yay!
```suggestion
  _pc.word = addr;
  _in_jump_state = true;
```